### PR TITLE
Track all event objects for a suite in JSON logger (7.0)

### LINKED ISSUE #33: JSON format in report.json not valid with --json option
See [#5023](https://github.com/Codeception/Codeception/issues/5023)

### PR DESCRIPTION
**Pull request by mycarrysun**
Fixes #33 and [#5203](https://github.com/Codeception/Codeception/issues/5023)

### CODE REVIEW

File: src/phpunit5-loggers.php [hunk 1]
```diff
@@ -290,34 +301,44 @@ public function endTest(\PHPUnit\Framework\Test $test, float $time) : void
...
+                        'output' => $output,
                     ]
                 );
             }
 
             /**
-             * @param string $buffer
+             * @param array $event_data
+             */
+            protected function addLogEvent($event_data = []): void
```
Comment (mycarrysun) [inline src/phpunit5-loggers.php:328]: Used a class prop to store all events so we can write all the events properly JSON encoded at the end of the suite

**Comment by DavertMik** [comment]
Got it, thanks!

**Comment by DavertMik** [comment]
Btw, where do you use JSON reports?

**Comment by mycarrysun** [comment]
@DavertMik to save the output to our database - very helpful for logging and looking at the event objects to see what passed etc...but before the PR the json_decode function returned null.

**Pull request merged**